Kitodo/src/main/java/de/sub/goobi/metadaten/Metadaten.java

```diff
@@ -1260,6 +1322,27 @@ public String addNode() throws TypeNotAllowedForParentException, TypeNotAllowedA
         return this.metaHelper.getAddableDocStructTypen(this.docStruct, true);
     }
 
+    /**
+     * Returns possible DocStruct types which can be added to current DocStruct.
+     *
+     * @return The DocStruct types.
+     */
+    public SelectItem[] getAddableDocStructTypes() {
+        switch (positionOfNewDocStrucElement) {
+            case BEFOR_CURRENT_ELEMENT:
+            case AFTER_CURRENT_ELEMENT:
+                return this.metaHelper.getAddableDocStructTypen(this.docStruct, true);
+
+            case FIRST_CHILD_OF_CURRENT_ELEMENT:
+            case LAST_CHILD_OF_CURRENT_ELEMENT:
+                return this.metaHelper.getAddableDocStructTypen(this.docStruct, false);
+
+            default:
+                logger.error("Invalid positionOfNewDocStrucElement");
+                return null;
```

Comment:
Returning null should be avoided as it introduce unnecessary null pointer checks. Could not a empty object be returned?